### ISSUE TITLE
Discussion: consider systemd services for our "daemons"

### ISSUE THREAD

**Issue by peterbarker** [ArduPilot/companion#17]
Currently we poke some "autostart" scripts directly into `/etc/rc.local`.

Modern Linux systems often use systemd services to help ensure startup ordering, restart-on-failure and clean shutdown on system stop.

We should consider using systemd services in place of rc.local.  This issue should hold arguments one way and the other so at some stage we can move to them or decide not to based on the balance of the arguments.

Some reasons NOT to move to systemd services:
 - if the system we wish to install on DOESN'T use systemd we'll need something roughly equivalent; rc.local or upstart configuration or similar
 - makes it less understandable by the average user (most people will understand how to add things to rc.local with a few minutes tinkering.  Adding systemd services is a little trickier.
 - both MAVProxy and cmavnode have useful consoles; we would need to run them in screen to keep that utility

**Comment by khancyr** [comment]
I don't think that systemd is harder than rc.local (disclamer : I don't know very much about those things, but I got some experience by using them). It is 4-5 line and give start/stop command ! 
My main consern is about the presence of systemd on rasberry pi ?
